source/agora/consensus/data/Block.d

```diff
@@ -459,13 +461,27 @@ public Block makeNewBlock (Transactions)(const ref Block prev_block,
     return block;
 }
 
-/// only used in unittests
+/// only used in unittests with defualt block signing with the genesis validators
 version (unittest)
 {
-    public Block makeNewBlock (Transactions)(const ref Block prev_block,
-        Transactions txs) @safe nothrow
+    public Block makeNewTestBlock (Transactions)(const ref Block prev_block,
+        Transactions txs, Enrollment[] enrollments = null, ulong cycle = 0) @safe nothrow
     {
-        return makeNewBlock(prev_block, txs, null);
+        return makeNewSignedTestBlock(prev_block, txs, null, cycle);
+    }
+}
+
+version (unittest)
+{
+    import agora.utils.Test : WK;
+
+    public Block makeNewSignedTestBlock (Transactions)(const ref Block prev_block,
+        Transactions txs, Enrollment[] enrollments = null, ulong cycle = 0,
+            KeyPair[] keys = [WK.Keys.NODE2, WK.Keys.NODE3, WK.Keys.NODE4,
+            WK.Keys.NODE5, WK.Keys.NODE6, WK.Keys.NODE7]) @safe nothrow
+    {
+        auto block = makeNewBlock(prev_block, txs, enrollments);
```

Comment:
Shouldn't this call `makeNewTestBlock`?

Reply:
No this is called from makeNewTestBlock after adding the block signature and validators mitmask to the block.

Reply:
Now resolved by removing other function.